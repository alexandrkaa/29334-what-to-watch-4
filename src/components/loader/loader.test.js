import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './loader.jsx';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

it(`<Loader /> should match snapshot`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus: `AUTH`,
      userData: {
        id: 1,
        email: `qwe@qwe.ru`,
        name: `qwe`,
        avatarUrl: `/img/avatar.jpg`
      }
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Loader />
          </BrowserRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
