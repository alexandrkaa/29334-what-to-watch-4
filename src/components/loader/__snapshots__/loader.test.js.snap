// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        aria-current="page"
        className="logo__link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="/img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>,
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "calc(100vh - 94px - 138px)",
        "justifyContent": "center",
        "width": "100vw",
      }
    }
  >
    <svg
      display="block"
      height="200"
      preserveAspectRatio="xMidYMid"
      style={
        Object {
          "margin": "auto",
        }
      }
      viewBox="0 0 100 100"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M85.141 70.133a40 40 0 00-71.194-36.487A42 40-62.865 0185.14 70.133"
        fill="#e15b64"
      >
        <animateTransform
          attributeName="transform"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          type="rotate"
          values="0 50 51;360 50 51"
        />
      </path>
    </svg>
  </div>,
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>,
]
`;
